core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -836,11 +844,6 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
     startStackIndex: Int,
     currentDepth: Int
   ): Exit[Any, Any] = {
-    assert(running.get)
```

Comment:
I removed this since we're already checking that we're running prior to calling this method, that way we avoid calling it repeatedly.

Or should it be added back as a safeguard in case we introduce a bug that sets the flag to `false` while the fiber is still running?

Reply:
I think it only has overhead if assertions are enabled for the JVM (albeit that might be all the time). It's mainly designed for bug detection.

Reply:
I'll add it back 👍

Reply:
By the way I just had a look at `assert`; it seems that assertion generation is controlled during compile-time, not at the JVM level. From `assert` scaladoc:

>  A set of `assert` functions are provided for use as a way to document
    and dynamically check invariants in code. Invocations of `assert` can be elided
    at compile time by providing the command line option `-Xdisable-assertions`,
    which raises `-Xelide-below` above `elidable.ASSERTION`, to the `scalac` command.

We should probably use the `-Xdisable-assertions` compiler flag when we generate the published artifacts, but that's probably better done in a separate PR.

Reply:
Great idea!